product-service/api/productApi.ts

```diff
@@ -1,88 +1,53 @@
-const products = [
-    {
-        "id": "1",
-        "name": "Bristlecone",
-        "manufacture": "Google",
-        "qubits": 72,
-        "price": 23000000
-    },
-    {
-        "id": "2",
-        "name": "Sycamore",
-        "manufacture": "Google",
-        "qubits": 54,
-        "price": 19000000
-    },
-    {
-        "id": "3",
-        "name": "Tenerife",
-        "manufacture": "IBM",
-        "qubits": 5,
-        "price": 1000000
-    },
-    {
-        "id": "4",
-        "name": "Yorktown",
-        "manufacture": "IBM",
-        "qubits": 5,
-        "price": 1000000
-    },
-    {
-        "id": "5",
-        "name": "Melbourne",
-        "manufacture": "IBM",
-        "qubits": 14,
-        "price": 3000000
-    },
-    {
-        "id": "6",
-        "name": "Rüschlikon",
-        "manufacture": "IBM",
-        "qubits": 16,
-        "price": 3200000
-    },
-    {
-        "id": "7",
-        "name": "Tokyo",
-        "manufacture": "IBM",
-        "qubits": 20,
-        "price": 4000000
-    },
-    {
-        "id": "8",
-        "name": "Austin",
-        "manufacture": "IBM",
-        "qubits": 20,
-        "price": 4000000
-    },
-    {
-        "id": "9",
-        "name": "Acorn",
-        "manufacture": "Rigetti",
-        "qubits": 19,
-        "price": 3800000
-    },
-    {
-        "id": "10",
-        "name": "Aspen-1",
-        "manufacture": "Rigetti",
-        "qubits": 16,
-        "price": 3100000
-    },
-    {
-        "id": "11",
-        "name": "Agave",
-        "manufacture": "Rigetti",
-        "qubits": 8,
-        "price": 2300000
-    },
-    {
-        "id": "12",
-        "name": "Tangle Lake",
-        "manufacture": "Intel",
-        "qubits": 49,
-        "price": 18000000
-    }
-];
+const { Client } = require('pg');
+
+const {PG_HOST, PG_PORT, PG_DATABASE, PG_USERNAME, PG_PASSWORD} = process.env;
+const dbOptions = {
+    host: PG_HOST,
+    port: PG_PORT,
+    database: PG_DATABASE,
+    user: PG_USERNAME,
+    password: PG_PASSWORD,
+    ssl: {
+        rejectUnauthorized: false
+    },
+    connectionTimeoutMillis: 5000
+};
 
-export const fetchProducts = async () => Promise.resolve(products);
\ No newline at end of file
+export const importProduct = async (productDetails) => {
+    return runQueryOnDD(`
+        begin;
+        insert into products (title, description, price) values
+        ('${productDetails.title}', '${productDetails.description}', ${productDetails.price});
```

Comment:
Please use Parameterized query to pass params to query.

https://node-postgres.com/features/queries - **Parameterized query** section

Reply:
Is it the best practice? I don't have a huge experience with DB. That's why I chose the most natural way to do it - template literals.

Reply:
yes, it's at least better than you have :)